### PR TITLE
[feature](insert)add hive table sink definition

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

issue: #31442

define hive table sink thrift and framework.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/PlanType.java [hunk 1]
```diff
@@ -95,6 +95,7 @@ public enum PlanType {
     // logical sinks
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/PlanType.java:95]:
```suggestion
    // physical sinks
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalHiveTableSink.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * logical olap table sink for insert command
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalHiveTableSink.java:42]: hive table

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHiveTableSink.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import org.apache.doris.nereids.util.Utils;
+import org.apache.doris.statistics.Statistics;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/** abstract physical sink */
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHiveTableSink.java:41]: hive sink

**Comment by wsjz** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**